sql-plugin/src/main/31xdb/scala/com/nvidia/spark/rapids/shims/Spark31XdbShimsBase.scala

```diff
@@ -135,6 +138,106 @@ trait Spark31XdbShimsBase extends SparkShims {
           TypeSig.DECIMAL_128),
         (a, conf, p, r) => new UnaryExprMeta[PromotePrecision](a, conf, p, r) {
           override def convertToGpu(child: Expression): GpuExpression = GpuPromotePrecision(child)
+        }),
+      expr[CheckOverflow](
```

Comment:
This is the same code that we had for the non-databricks version and is adding 100 lines of copy/pased code. I would much rather have us use the new way of doing shims, like was mentioned in the [comment here](https://github.com/NVIDIA/spark-rapids/pull/6937#pullrequestreview-1160234172)

Reply:
I have made the changes after an offline discussion

A couple of things that convinced me was that even though the way this change was inheriting is also the new way of doing the shims, what's different about the suggested way is that it gives us more of a `hasA` instead of an `isA` flexibility. We could've still kept this tagging on inside the `SparkShimImpl.getExprs` but we want to ultimately get rid of the `SparkShimImpl.getExprs` and go for a more granular control in the `GpuOverrides` to what exprs are being overriden.